notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/ProducerNotificationEventUtils.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (C) 2018-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.core.events;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import com.expediagroup.streamplatform.streamregistry.avro.AvroEvent;
+import com.expediagroup.streamplatform.streamregistry.avro.AvroKey;
+import com.expediagroup.streamplatform.streamregistry.avro.AvroKeyType;
+import com.expediagroup.streamplatform.streamregistry.avro.AvroProducer;
+import com.expediagroup.streamplatform.streamregistry.model.Producer;
+import com.expediagroup.streamplatform.streamregistry.model.Status;
+
+@Slf4j
+public class ProducerNotificationEventUtils {
+
+  public static AvroKey toAvroKeyRecord(Producer producer) {
+    validateProducerKey(producer);
+    val key = producer.getKey();
+    val producerName = key.getName();
+    val streamName = key.getStreamName();
+    val streamVersion = key.getStreamVersion();
+    val streamDomain = key.getStreamDomain();
+    val zone = key.getZone();
+
+    var domainKey = AvroKey.newBuilder()
+        .setId(streamDomain)
+        .setType(AvroKeyType.DOMAIN)
+        .build();
+
+    var zoneKey = AvroKey.newBuilder()
+        .setId(zone)
+        .setParent(domainKey)
+        .setType(AvroKeyType.ZONE)
+        .build();
+
+    var streamKey = AvroKey.newBuilder()
+        .setId(streamName)
+        .setParent(zoneKey)
+        .setType(AvroKeyType.STREAM)
+        .build();
+
+    var streamVersionKey = AvroKey.newBuilder()
+        .setId(streamVersion.toString())
+        .setParent(streamKey)
+        .setType(AvroKeyType.STREAM_VERSION)
+        .build();
+
+    return AvroKey.newBuilder()
+        .setId(producerName)
+        .setParent(streamVersionKey)
+        .setType(AvroKeyType.PRODUCER)
```

Comment:
now that we have lineage from both sides you could set the physical for a producer to the zone?